extension/chrome/src/js/userhosts.js

```diff
@@ -1,27 +1,35 @@
 document.getElementById('save_user_hosts').addEventListener('click', (event) => {
-	
-	let hostsButton = document.getElementById('user_defined_hosts'),
-		hosts = hostsButton.value.
+
+    let hostsTextarea = document.getElementById('user_defined_hosts');
+    let hosts = hostsTextarea.value.
 		replace(/,/g, ' ').
 		replace(/;/g, ' ').
-		replace(/\r/g, ' ').
-		replace(/\t/g, ' ').
-		replace(/\n/g, ' ').
-		split(' ');
-
-	let fhosts = [];
+		replace(/\t+/g, ' ').
+		replace(/\r+/g, ' ').
+        replace(/\n+/g, ' ').
+        trim().
+        split(' ').
+        filter(host => {
```

Comment:
`.filter(Boolean)` could be used.
or `.filter(host => !!host)`, no need to wrap it in {}